src/middleware/header.js

```diff
@@ -131,8 +139,45 @@ module.exports = function (middleware) {
 
 				setBootswatchCSS(templateValues, res.locals.config);
 
+				var unreadCount = {
+					topic: results.unreadTopicCount || 0,
+					newTopic: results.unreadNewTopicCount || 0,
+					watchedTopic: results.unreadWatchedTopicCount || 0,
+					chat: results.unreadChatCount || 0,
+					notification: results.unreadNotificationCount || 0,
+				};
+				Object.keys(unreadCount).forEach(function (key) {
+					if (unreadCount[key] > 99) {
+						unreadCount[key] = '99+';
+					}
+				});
+
+				results.navigation = results.navigation.map(function (item) {
+					if (item.originalRoute === '/unread' && results.unreadTopicCount > 0) {
+						return Object.assign({}, item, {
+							content: unreadCount.topic,
+							iconClass: item.iconClass + ' unread-count',
+						});
+					}
+					if (item.originalRoute === '/unread/new' && results.unreadNewTopicCount > 0) {
+						return Object.assign({}, item, {
+							content: unreadCount.newTopic,
+							iconClass: item.iconClass + ' unread-count',
+						});
+					}
+					if (item.originalRoute === '/unread/watched' && results.unreadWatchedTopicCount > 0) {
+						return Object.assign({}, item, {
+							content: unreadCount.watchedTopic,
+							iconClass: item.iconClass + ' unread-count',
+						});
+					}
```

Comment:
Consider using a `switch (item.originalRoute) { ... };` here

Reply:
Still need the `if`s for checking `> 0` so this is shorter.